_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -594,6 +611,22 @@ public function update_data( $request ) {
 			$value = Jetpack_Core_Json_Api_Endpoints::cast_value( $value, $option_attrs );
 
 			switch ( $option ) {
+				case 'lang_id':
+					if ( defined( 'WPLANG' ) || ! current_user_can( 'install_languages' ) ) {
+						// We can't affect this setting
+						$updated = false;
+						break;
+					}
+
+					if ( $value === 'en_US' || empty( $value ) ) {
+						return delete_option( 'WPLANG' );
+					}
+
+					// `wp_download_language_pack` only tries to download packs if they're not already available
+					$language = wp_download_language_pack( $value );
```

Comment:
Todo: make sure `wp_download_language_pack` doesn't return `false` before setting the option.